### PR TITLE
Uniformize coding style

### PR DESCRIPTION
**Pull request by AntonioND**
It has always bothered me that all the files have different coding styles (and there are many inconsistencies even within files). I've wanted to do a cleanup like this for a long time. I know that this is a problem for whoever has non-upstream patches, but I still think it's something that had to be done. After all, this toolchain has had many contributors since the 90's, so it's normal that they all have slightly different styles. This doesn't fix all inconsistencies, but it's something.

It's difficult to agree on a coding style, so I've decided to use the script that they use for the Linux kernel: https://github.com/torvalds/linux/blob/master/scripts/checkpatch.pl

I've added 2 targets to the Makefile:
- `checkpatch` does a check of all patches between the current HEAD and the first common commit between HEAD and origin/develop.
- `checkcodebase` analyzes all the files of the codebase.

### CODE REVIEW

File: .checkpatch.conf [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# Allow the usage of 'extern' in .c files
+--ignore AVOID_EXTERNS
+
+# There's no BIT macro
+--ignore BIT_MACRO
+
+# Allow CamelCase
+--ignore CAMELCASE
+
+# Comparing to NULL explicitely isn't a bad thing
```
Comment (Ben10do) [inline .checkpatch.conf]: *explicitly!

**Comment by Ben10do** [comment]
Not sure what Travis is up to, but I can confirm that this builds properly on macOS. I've not tried the `checkpatch` target, but I see no reason why that shouldn't work!

File: .checkpatch.conf (hunk 1 above)
Comment (AntonioND) [reply .checkpatch.conf]: Done!

**Comment by AntonioND** [comment]
Yes, I noticed that the macOS build was failing because of some infrastructure problem, it seems it's working now...

If there are suggestions, it's fine. I don't like all the rules (and I've had to disable a few of them because the resulting code was unreadable in some cases). However, it's easier to follow this style because of the `checkpatch.pl` script.

I was thinking about making Travis CI complain if there are ERRORs in checkpatch. WARNINGs and CHECKs are just "nice to have" things, but they generally help (at least the WARNINGs). I was also trying to find a way of sending a message to the PR with the result of the script, but that is secondary.

**Comment by AntonioND** [comment]
Also, this doesn't mean that the rules can't change. For example, I'd like to enable `AVOID_EXTERNS` at some point, but that requires a bit more of refactoring than what I wanted to do.

**Pull request merged**